desktop_version/src/GraphicsResources.cpp

```diff
@@ -116,21 +116,32 @@ void GraphicsResources::init(void)
 
 	auto templist = FILESYSTEM_getGraphicsDirFileNames();
 	for (auto name : templist) {
-		if (name.find("graphics/tiles") != 0 || name.length() < 4 || name.substr(name.length()-4, 4) != ".png")
-			continue;
+		printf("%s\n", name.c_str());
+		if (name.find("graphics/tiles") == 0) {
+			if (name.substr(name.length()-4, 4) != ".png") continue;
```

Comment:
You removed the `name.length() < 4` sanity check, which has to come before the `name.substr(name.length()-4, 4)` to prevent that check from running by way of short-circuit. Otherwise, if the length of the name is less than 4, `name.length()-4` will index out-of-bounds, which is undefined behavior.

Reply:
The `name.length() < 4` sanity check seems redundant, since in order for the code to reach `name.substr(name.length()-4, 4)` it must have passed `name.find("graphics/tiles") == 0` / `name.find("graphics/sprites") == 0` and therefor be long enough.

Reply:
Makes sense to me, then.